File: exercises/day13-Keyboard-First-Refactoring/csharp/SleighTripService/Trip/SleighTripDAO.cs
using SleighTripService.Exception;

namespace SleighTripService.Trip;

/// <summary>
/// Legacy static data access object for sleigh trips.
/// </summary>
public static class SleighTripDAO
{
    public static List<SleightTrip> FindTripsByUser(Elf.Elf user)
    {
        throw new CollaboratorCallException(
            "SleighTripDAO is a static collaborator and should not be used directly in unit tests"
        );
    }
}

File: exercises/day13-Keyboard-First-Refactoring/typescript/src/trip/SleighTripDAO.ts
import { SleightTrip } from './SleightTrip';
import { Elf } from '../elf/Elf';
import { CollaboratorCallException } from '../exception/Exceptions';

/**
 * Legacy static data access object for sleigh trips.
 */
export class SleighTripDAO {
    static findTripsByUser(user: Elf): SleightTrip[] {
        throw new CollaboratorCallException(
            'SleighTripDAO is a static collaborator and should not be used directly in unit tests'
        );
    }
}
